### PR TITLE
Game book in English

### PR DESCRIPTION
**Pull request by LuigiImVector**
## What does this PR do?
Add resource: game development (Language agnostic) book

### How do we know it's really free?
The book is published on the author's official website

### For book lists, is it a book? For course lists, is it a course? etc.
It is a book

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [X] Search for duplicates.
- [X] Include author(s) and platform where appropriate.
- [X] Put lists in alphabetical order, correct spacing.
- [X] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by eshellman** [comment]
great resource, but our game development resources are currently in "misc". there are certainly enough of them to support a new category. Would be great to create the new category, with other appropriate books. If you don't feel up to moving them, to the new category, put this in misc and we can leave it to another PR to create the category.

**Comment by LuigiImVector** [comment]
I transferred 6 books from the "Misc" category to the new "Game Development" category

File: books/free-programming-books.md [hunk 1]
```diff
@@ -467,6 +468,17 @@
 * [Theory and Applications for Advanced Text Mining](http://www.intechopen.com/books/theory-and-applications-for-advanced-text-mining)
 
 
+#### Game Development
+
+* [2D Game Development: From Zero To Hero](https://github.com/Penaz91/2DGD_F0TH) (:construction: *in process*)
```
Comment (davorpa) [inline books/free-programming-books.md:473]: Maybe better use deployed url seen in repo?

https://therealpenaz91.itch.io/2dgd-f0th

File: books/free-programming-books.md [hunk 2]
```diff
@@ -467,6 +468,17 @@
...
+* [Coding With Minecraft](https://turtleappstore.com/book/) - Al Sweigart
+* [Designing Virtual Worlds](https://mud.co.uk/richard/DesigningVirtualWorlds.pdf) - Richard A. Bartle (PDF)
+* [Game AI Pro](http://www.gameaipro.com) - Steve Rabin
+* [Game Design with AGS](https://ensadi.github.io/AGSBook/) - Dave Ensminger, A. G. Madi
+* [Game Programming Patterns](http://gameprogrammingpatterns.com) - Bob Nystrom
+* [Procedural Content Generation in Games](https://pcgbook.com) - Noor Shaker, Julian Togelius, Mark Nelson
```
Comment (davorpa) [inline books/free-programming-books.md:479]: Browser raised `NET::ERR_CERT_AUTHORITY_INVALID`

Maybe change to `http` protocol

```suggestion
* [Procedural Content Generation in Games](http://pcgbook.com) - Noor Shaker, Julian Togelius, Mark Nelson
```

File: books/free-programming-books.md (hunk 1 above)
Comment (LuigiImVector) [reply books/free-programming-books.md:473]: I think it's not necessary, in the repository there are some information that there aren't in the other site. I let the staff decide what to do.

**Comment by eshellman** [comment]
IIsn't this past the "*in process*" phase?

File: books/free-programming-books.md (hunk 1 above)
Comment (davorpa) [reply books/free-programming-books.md:473]: Ok. Let @Penaz91 decide and if needed threat in other issue.

Comment (Penaz91) [reply books/free-programming-books.md:473]: The ebook has two main repositories where the sources can be found (both on Gitlab and Github) but the epub and pdf versions are published on itch just for ease of deployment. Feel free to reference any url you wish.

**Comment by Penaz91** [comment]
> IIsn't this past the "_in process_" phase?

Due to the nature of the project itself (it's meant to be some kind of "Evolving Body of knowledge"), the book could be seen as "always in process". As of now it features about 400 pages of content but it still has a lot of placeholders that I think could frustrate some readers. Sadly I'm on my own in writing this book, so progress has slowed down considerably, hopefully this isn't giving you trouble.

Again, thanks for the help and letting this book be in your repository.

**Pull request merged**